File: Atividades/atividade-pratica-01/src/main/java/com/example/bloodDonationSchedule/entity/Pessoa.java
package com.example.bloodDonationSchedule.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "pessoas")
public class Pessoa {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column
    private String nome;
    @Column
    private String rua;
    @Column
    private String numero;
    @Column
    private String complemento;
    @Column
    private String rg;
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "cidade_id")
    private Cidade cidade;
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "tipo_id")
    private TipoSanguineo tipoSanguineo;
    @Column
    private LocalDateTime created_at;
    @Column
    private LocalDateTime updated_at;
}

File: Atividades/atividade-pratica-02/src/services/pessoa.js
import { api } from "./api";

export const getPessoas = async () => {
  const response = await api.get("pessoas");
  return response.data;
};

export const postNovaPessoa = async (
  nome,
  rua,
  numero,
  complemento,
  rg,
  cidade_id,
  tipoSanguineo_id
) => {
  const response = await api.post("pessoas", {
    nome,
    rua,
    numero,
    complemento,
    rg,
    cidade_id,
    tipoSanguineo_id,
  });
  return response.data;
};

export const patchPessoa = async (
  id,
  nome,
  rua,
  numero,
  complemento,
  rg,
  cidade_id,
  tipoSanguineo_id
) => {
  const response = await api.patch(`pessoas/${id}`, {
    nome,
    rua,
    numero,
    complemento,
    rg,
    cidade_id,
    tipoSanguineo_id,
  });
  return response.data;
};

export const deletePessoa = async (id) => {
  const response = await api.delete(`pessoas/${id}`);
  return response.data;
};
